planner/core/planbuilder.go

```diff
@@ -1992,7 +2056,101 @@ func (b *PlanBuilder) buildAnalyzeFullSamplingTask(
 		}
 		taskSlice = append(taskSlice, newTask)
 	}
-	return taskSlice, nil
+	return taskSlice, globalV2AnalyzeOptions, nil
+}
+
+func (b *PlanBuilder) getSavedAnalyzeOpts(physicalID int64, tblInfo *model.TableInfo) (map[ast.AnalyzeOptionType]uint64, model.ColumnChoice, []*model.ColumnInfo, error) {
+	analyzeOptions := map[ast.AnalyzeOptionType]uint64{}
+	if !variable.PersistAnalyzeOptions.Load() {
+		return analyzeOptions, model.DefaultChoice, nil, nil
+	}
+	exec := b.ctx.(sqlexec.RestrictedSQLExecutor)
+	stmt, err := exec.ParseWithParams(context.TODO(), "select * from mysql.analyze_options where table_id = %?", physicalID)
```

Comment:
Maybe it is better to specify select fields, which may improve code readability.

Reply:
Done. Also to avoid issues when adding new columns.